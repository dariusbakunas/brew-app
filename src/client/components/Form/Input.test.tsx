import { shallow } from 'enzyme';
import React from 'react';

import { Form } from '../index';

describe('Input', () => {
  it('should render without throwing an error', () => {
    const wrapper = shallow(<Form.Input id="test-input" label="test" name="test" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onChange handler on user input', () => {
    const handler = jest.fn();
    const wrapper = shallow(<Form.Input label="test" name="test" onChange={handler} />);
    const event = { currentTarget: { value: '123456789abcde' } };
    wrapper.find('input').simulate('change', event);
    expect(handler).toHaveBeenCalledWith(event, { name: 'test', value: '123456789abcde' });
  });

  it('should render with icon', () => {
    const wrapper = shallow(<Form.Input id="test-input" label="test" name="test" icon="percent" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should convert input to number if type is number', () => {
    const handler = jest.fn();
    const wrapper = shallow(<Form.Input label="test" name="test" type="number" step={0.1} onChange={handler} />);
    const event = { currentTarget: { value: '325.6' } };
    wrapper.find('input').simulate('change', event);
    expect(handler).toHaveBeenCalledWith(event, { name: 'test', value: 325.6 });
  });

  it('should render error span if error is specified', () => {
    const wrapper = shallow(<Form.Input label="test" name="test" icon="percent" error="Test error" />);
    expect(wrapper.find('input').hasClass('uk-form-danger')).toBeTruthy();
    expect(wrapper.find('.uk-text-danger')).toHaveLength(1);
    expect(wrapper.find('.uk-text-danger').text()).toEqual('Test error');
  });
});
