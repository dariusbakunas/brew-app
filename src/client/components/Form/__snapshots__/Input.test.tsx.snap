// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render with icon 1`] = `
<Fragment>
  <label
    className="uk-form-label"
    htmlFor="test-input"
  >
    test
  </label>
  <div
    className="uk-inline"
  >
    <span
      className="uk-form-icon uk-form-icon-flip uk-icon"
    >
      <Icon
        icon="percent"
        width="20px"
      />
    </span>
    <input
      className="uk-input"
      id="test-input"
      name="test"
      onChange={[Function]}
      step={null}
      type="text"
    />
  </div>
</Fragment>
`;

exports[`Input should render without throwing an error 1`] = `
<Fragment>
  <label
    className="uk-form-label"
    htmlFor="test-input"
  >
    test
  </label>
  <input
    className="uk-input"
    id="test-input"
    name="test"
    onChange={[Function]}
    step={null}
    type="text"
  />
</Fragment>
`;
